sdk/spring-sdk/src/test/java/kalix/springsdk/testmodels/view/ViewTestModels.java

```diff
@@ -149,6 +171,122 @@ public TransformedUser getUser(@RequestBody ByEmail byEmail) {
+   */
+  @Table("users_view")
+  @Subscribe.ValueEntity(UserEntity.class)
+  public static class ViewWithSubscriptionsInMixedLevelsHandleDelete extends View<TransformedUser> {
+
+    @Subscribe.ValueEntity.HandleDeletes(UserEntity.class)
+    public UpdateEffect<TransformedUser> onDelete() {
+      return effects().deleteState();
+    }
+
+    @Query("SELECT * FROM users_view WHERE email = :email")
+    @PostMapping("/users/by-email")
+    public TransformedUser getUser(@RequestBody ByEmail byEmail) {
+      return null;
+    }
+  }
+
+  @Table("users_view")
+  public static class ViewWithoutSubscriptionButWithHandleDelete extends View<TransformedUser> {
+
+    @Subscribe.ValueEntity.HandleDeletes(UserEntity.class)
+    public UpdateEffect<TransformedUser> onDelete() {
+      return effects().deleteState();
+    }
+
+    @Query("SELECT * FROM users_view WHERE email = :email")
+    @PostMapping("/users/by-email")
+    public TransformedUser getUser(@RequestBody ByEmail byEmail) {
+      return null;
+    }
+  }
+
+  @Table("users_view")
+  public static class ViewDuplicatedHandleDeletesAnnotations extends View<TransformedUser> {
+
+    @Subscribe.ValueEntity(UserEntity.class)
+    public UpdateEffect<TransformedUser> onChange(User user) {
+      return effects()
+          .updateState(new TransformedUser(user.lastName + ", " + user.firstName, user.email));
+    }
+
+    @Subscribe.ValueEntity.HandleDeletes(UserEntity.class)
+    public UpdateEffect<TransformedUser> onDelete() {
+      return effects().deleteState();
+    }
+
+    @Subscribe.ValueEntity.HandleDeletes(UserEntity.class)
+    public UpdateEffect<TransformedUser> onDelete2() {
+      return effects().deleteState();
+    }
+
+    @Query("SELECT * FROM users_view WHERE email = :email")
+    @PostMapping("/users/by-email")
+    public TransformedUser getUser(@RequestBody ByEmail byEmail) {
+      return null;
+    }
+  }
+
+  @Table("users_view")
+  public static class ViewDuplicatedSubscriptions extends View<TransformedUser> {
+
+    @Subscribe.ValueEntity(UserEntity.class)
+    public UpdateEffect<TransformedUser> onChange(User user) {
+      return effects()
+          .updateState(new TransformedUser(user.lastName + ", " + user.firstName, user.email));
+    }
+
+    @Subscribe.ValueEntity(UserEntity.class)
+    public UpdateEffect<TransformedUser> onChange2(User user) {
+      return effects()
+          .updateState(new TransformedUser(user.lastName + ", " + user.firstName, user.email));
+    }
+
+    @Subscribe.ValueEntity.HandleDeletes(UserEntity.class)
+    public UpdateEffect<TransformedUser> onDelete() {
+      return effects().deleteState();
+    }
+
+    @Query("SELECT * FROM users_view WHERE email = :email")
+    @PostMapping("/users/by-email")
+    public TransformedUser getUser(@RequestBody ByEmail byEmail) {
+      return null;
+    }
+  }
+
+  @Table("users_view")
+  public static class ViewWithMissingSubscriptionForHandleDeletes extends View<TransformedUser> {
```

Comment:
We can drop this case for now.

When we allow multiple subscriptions, then we will need a new set of rules. 

* All subscriptions must be at method level when subscribing to more than one entity. 
* The number of allowed `HandleDeletes` should equal the number of unique value entity types that are being subscribed to.
* There must be no more than one `HandleDelete` per entity (that's a relevant check already)